### PR TITLE
core: use ory-am/hydra SDK

### PR DESCRIPTION
**Pull request by faxal**

### CODE REVIEW

File: core/idp.go [hunk 1]
```diff
@@ -91,85 +89,35 @@ func (idp *IDP) refreshCache(key string) {
...
 	if err != nil {
 		return nil, err
 	}
 
-	key, err := jwk.DecodePublicKey()
-	if err != nil {
-		return nil, err
+	rsaKey, ok := hjwk.First(jwk.Keys).Key.(*rsa.PublicKey)
+	if !ok {
+		return nil, errors.New("Could not convert to RSA Public Key")
```
Comment (janekolszak) [inline core/idp.go]: create an error in error.go

Comment (faxal) [inline core/idp.go]: done!

**Comment by janekolszak** [comment]
What's wrong with package `github.com/ory-am/hydra/sdk` that you haven't used that?

I'll accept this PR, but next time please let us know what are you working on. We duplicated the work.

**Comment by faxal** [comment]
Sorry about that. Will do next time.

**Comment by faxal** [comment]
I don't understand this part: 

> What's wrong with package github.com/ory-am/hydra/sdk that you haven't used that?

**Comment by janekolszak** [comment]
Sorry, I thought hjwk.First() was in skd's API as well.

**Pull request merged**